File: Python/lab4/interfaces/Factory.py
from abc import ABCMeta, abstractmethod
from interfaces.LaptopProduct import LaptopProduct
from interfaces.PcProduct import PcProduct
from interfaces.PhoneProduct import PhoneProduct

class Factory(metaclass=ABCMeta):
    @abstractmethod
    def createLaptop(self, type_of_laptop) -> LaptopProduct:
        pass

    @abstractmethod
    def createPc(self, type_of_pc) -> PcProduct:
        pass

    @abstractmethod
    def createPhone(self, type_of_phone) -> PhoneProduct:
        pass

File: Java/lab4/src/main/java/org/example/interfaces/Factory.java
package org.example.interfaces;

import org.example.factory.PcFactory;
import org.example.factory.PhoneFactory;
public interface Factory {
    LaptopProduct createLaptop(String typeOfLaptop);
    PcProduct createPc(String typeOfPc);
    PhoneProduct createPhone(String typeOfPhone);
}
